cli/commands.go

```diff
@@ -437,11 +437,46 @@ func ImportImage(c *cli.Context) error {
 
 // AddNode adds a node to an existing cluster
 func AddNode(c *cli.Context) error {
+
+	/*
+	 * (0) Check flags
+	 */
+
 	clusterName := c.String("name")
-	nodeRole := c.String("role")
 	nodeCount := c.Int("count")
 
-	// check if cluster (i.e. the server) exists
+	/* (0.1)
+	 * --role
+	 * Role of the node that has to be created.
+	 * One of (server|master), (agent|worker)
+	 */
+	nodeRole := c.String("role")
+	if nodeRole == "worker" {
+		nodeRole = "agent"
+	}
+	if nodeRole == "master" {
+		nodeRole = "server"
+	}
+
+	// TODO: support this
+	if nodeRole == "server" {
+		return fmt.Errorf("ERROR: sorry, we don't support adding server nodes at the moment!")
+	}
+
+	/* (0.2)
+	 * --image, -i
+	 * The k3s image used for the k3d node containers
```

Comment:
It would be nice to use the image that currently runs the cluster by default...  If you agree in principle, we can improve this laster.

Reply:
Follow up is fine.

Reply:
Follow up is fine.